### DISCUSSION TITLE
Failing to update few rabbitmq parameters ( Channel Operation Timeout , Consumer Prefetch )

### DISCUSSION THREAD

**Discussion by GitJadhav** [rabbitmq/rabbitmq-server#10927] [Questions]
### Describe the bug

RabbitMQ Version - 3.12.x
Trying to tune the following parameters running on defaults 
1.Channel Operation Timeout
sudo rabbitmqctl environment | grep "channel_operation_timeout"
      {channel_operation_timeout,15000},

2.Consumer Prefetch
sudo rabbitmqctl environment | grep "default_consumer_prefetch"
      {default_consumer_prefetch,{false,0}},

### Reproduction steps

1.Tune Channel Operation Timeout 
File - rabbitmq.conf 
channel_operation_timeout = 60000 
2.Tune consumer prefetch 
File - advanced.conf
[
{rabbit, [
       {default_consumer_prefetch,{false,250}}
     ]
 }
]
3.rabbitmqctl stop_app , rabbitmqctl start_app
...

### Expected behavior

1.Run "rabbitmqctl environment" 
2.Confirm the new settings for Channel Operation Timeout and Consumer Prefetch 
{channel_operation_timeout,60000}
{default_consumer_prefetch,{false,250}}

### Additional context

I am still seeing the defaults and not seeing any tuned settings being applied 
Run "rabbitmqctl environment" 
{channel_operation_timeout,15000},
{default_consumer_prefetch,{false,0}}

**Comment by lukebakken** [comment]
* Please _ATTACH_ the complete configuration files you are using to change these settings.
* Are you certain that RabbitMQ is using your configuration files? It will be logged on RabbitMQ startup.

**Comment by GitJadhav** [comment]
@lukebakken 
Attaching the config files 
[advanced.conf.json](https://github.com/rabbitmq/rabbitmq-server/files/14889108/advanced.conf.json)
[rabbitmq.conf.json](https://github.com/rabbitmq/rabbitmq-server/files/14889109/rabbitmq.conf.json)

For logging on RabbitMQ Startup -- Are yo referring to this logs 
sudo journalctl -xe
rabbit.log

**Reply by michaelklishin** [reply]
The correct name of the advanced config file [is not `advanced.conf`, it is `advanced.config`](https://www.rabbitmq.com/docs/configure#advanced-config-file).

`advanced.conf` will not be loaded.

**Comment by GitJadhav** [comment]
@michaelklishin 
I was not even able to start rabbit with advanced.config ..
[ec2-user@ip rabbitmq]$ ls -la
total 52
drwxr-sr-x.  2 root     rabbitmq    98 Apr  5 20:35 .
drwxr-xr-x. 87 root     root      8192 Apr  5 16:01 ..
-rw-r--r--.  1 rabbitmq rabbitmq   439 Apr  4 21:31 advanced.config
-rw-r--r--.  1 root     rabbitmq    23 Nov 27 21:11 enabled_plugins
-rw-r-----.  1 rabbitmq rabbitmq 27319 Apr  4 22:38 rabbitmq.conf
-rw-r--r--.  1 rabbitmq rabbitmq   161 Mar 28 19:56 rabbitmq-env.conf

[ec2-user@ip rabbitmq]$ sudo systemctl stop rabbitmq-server.service

[ec2-user@ip rabbitmq]$ sudo systemctl start rabbitmq-server.service
Job for rabbitmq-server.service failed because the control process exited with error code.
See "systemctl status rabbitmq-server.service" and "journalctl -xe" for details.
[ec2-user@ip rabbitmq]$ 
[ec2-user@ip rabbitmq]$ sudo systemctl status  rabbitmq-server.service
● rabbitmq-server.service - RabbitMQ broker
   Loaded: loaded (/usr/lib/systemd/system/rabbitmq-server.service; enabled; vendor preset: disabled)
  Drop-In: /etc/systemd/system/rabbitmq-server.service.d
           └─limits.conf
   Active: activating (auto-restart) (Result: exit-code) since Fri 2024-04-05 20:36:25 UTC; 8s ago
  Process: 15544 ExecStop=/usr/sbin/rabbitmqctl shutdown (code=exited, status=0/SUCCESS)
  Process: 15651 ExecStart=/usr/sbin/rabbitmq-server (code=exited, status=1/FAILURE)
 Main PID: 15651 (code=exited, status=1/FAILURE)
   Status: "Standing by"

**Reply by lukebakken** [reply]
More than likely the `advanced.config` file is not formatted correctly, or maybe saved with a UTF-8 BOM.

**Reply by lukebakken** [reply]
`#` is _NOT_ the comment character in `advanced.config`, `%` is.

**Reply by GitJadhav** [reply]
@lukebakken I took all the comment notes and just have the config parameter 
I am still unable to start the rabbitmq 
[ec2-user@ip rabbitmq]$ sudo systemctl start rabbitmq-server.service
Job for rabbitmq-server.service failed because the control process exited with error code.
See "systemctl status rabbitmq-server.service" and "journalctl -xe" for details.
[ec2-user@ip rabbitmq]$ 

Apr 05 21:50:10 ip.ec2.internal rabbitmq-server[30700]: BOOT FAILED
Apr 05 21:50:10 ip.ec2.internal rabbitmq-server[30700]: ===========
Apr 05 21:50:10 ip.ec2.internal rabbitmq-server[30700]: Error during startup: {error,failed_to_read_advanced_c>

**Reply by lukebakken** [reply]
`error,failed_to_read_advanced_c` - failed to read `advanced.config` file.

You should try starting `rabbitmq-server` in the foreground to get more information.
